usecases/modulecomponents/batch/batch.go

```diff
@@ -193,6 +210,7 @@ func (b *Batch) batchWorker(execAfterRequestFunction func(limits *modulecomponen
 		if len(texts) > 0 && objCounter == len(job.texts) {
 			_ = b.makeRequest(job, texts, job.cfg, origIndex, rateLimit)
 		}
+		rateLimitPerApiKey[job.apiKeyHash] = rateLimit
```

Comment:
is it true `apiKeyHash` is associated with each `RateLimit` object?

Reply:
yes - each Api key has an independent rate limit so we want to keep them separate from each other

Reply:
inside the the `RateLimits` struct, have a `hash` or `apiKey` hash field that would be passed together with `RateLimits` struct, if it would be empty then it would mean `unlimited`. That way only the modules where we want to turn on limiting would have to generate the `apieKeyHash` those that don't would not have to do aynthig.
This way we could remove `GetApiKeyHash` method from the interface so that it would not have to be mandatory for all modules.

Reply:
Sadly there are two problems:
- The rate limit is per-key. Concrete example: I used an cohere trial key with 5 reqs/min and then a production key with 10k reqs/min. The only way to reset the rate limit was to restart weaviate.
- Only the client-package knows the api-key. So the info must somehow come from that package.

I think the easiest way would be to add a `GetApiKeyHash()` method to local vectorizers that sinply returns an empty string. Not ideal, but if we want to share the code we need to support all possibilities :(

Reply:
Yeah, I know it's not ideal that's why I'm trying to explore other possibilities :) so that maybe we would come to some other idea, but OK I get it. Let's leave it then